corehq/form_processor/tests/utils.py

```diff
@@ -347,8 +348,11 @@ def create_case(
     if save:
         # disable publish to Kafka to avoid intermittent errors caused by
         # the nexus of kafka's consumer thread and freeze_time
-        with patch.object(FormProcessorSQL, "publish_changes_to_kafka"):
+        if enable_kafka:
             FormProcessorSQL.save_processed_models(ProcessedForms(form, None), [case])
+        else:
+            with patch.object(FormProcessorSQL, "publish_changes_to_kafka"):
+                FormProcessorSQL.save_processed_models(ProcessedForms(form, None), [case])
```

Comment:
nit: this could be refactored for a tiny bit less duplication
```suggestion
        from contextlib import nullcontext
        kafka_patch = patch.object(FormProcessorSQL, "publish_changes_to_kafka")
        with (nullcontext() if enable_kafka else kafka_patch):
            FormProcessorSQL.save_processed_models(ProcessedForms(form, None), [case])
```

Reply:
Ah thanks! I figured when I wrote that there was a less repetitive way, but forgot to dig into it.